CMakeLists.txt

```diff
@@ -11,21 +19,20 @@ option(KokkosFFT_INTERNAL_Kokkos "Build internal Kokkos instead of relying on ex
 option(KokkosFFT_ENABLE_BENCHMARK "Build benchmarks for KokkosFFT" OFF)
 
 # Version information
-set(KokkosFFT_VERSION_MAJOR 0)
-set(KokkosFFT_VERSION_MINOR 0)
-set(KokkosFFT_VERSION_PATCH 00)
-set(KokkosFFT_VERSION "${KokkosFFT_VERSION_MAJOR}.${KokkosFFT_VERSION_MINOR}.${KokkosFFT_VERSION_PATCH}")
-
-#Set variables for config file
-math(EXPR KOKKOSFFT_VERSION "${KokkosFFT_VERSION_MAJOR} * 10000 + ${KokkosFFT_VERSION_MINOR} * 100 + ${KokkosFFT_VERSION_PATCH}")
+math(EXPR KOKKOSFFT_VERSION "${PROJECT_VERSION_MAJOR} * 10000 + ${PROJECT_VERSION_MINOR} * 100 + ${PROJECT_VERSION_PATCH} * 10 + ${PROJECT_VERSION_TWEAK}")
 math(EXPR KOKKOSFFT_VERSION_MAJOR "${KOKKOSFFT_VERSION} / 10000")
 math(EXPR KOKKOSFFT_VERSION_MINOR "${KOKKOSFFT_VERSION} / 100 % 100")
 math(EXPR KOKKOSFFT_VERSION_PATCH "${KOKKOSFFT_VERSION} % 100")
```

Comment:
Do you really need to do that? You can use advantage of the standard version variables:

```cmake
set(KOKKOSFFT_VERSION "${PROJECT_VERSION}")
set(KOKKOSFFT_VERSION_MAJOR "${PROJECT_VERSION_MAJOR}")
set(KOKKOSFFT_VERSION_MINOR "${PROJECT_VERSION_MINOR}")
set(KOKKOSFFT_VERSION_PATCH "${PROJECT_VERSION_PATCH}")
```

Also, you specified a semver with tweak version (as of 0.0.0.0), and don't use it here.

Reply:
I have never used tweak version, but Kokkos version seems to have that. https://github.com/CExA-project/kokkos-fft/pull/47#discussion_r1485902008

The easiest solution is removing the tweak version.